code/modules/clothing/spacesuits/plasmamen.dm

```diff
@@ -134,6 +156,15 @@
 
 	update_action_buttons()
 
+/obj/item/clothing/head/helmet/space/plasmaman/attack_hand_secondary(mob/user)
+	. = ..()
+	if(src.attached_hat)
+		user.put_in_active_hand(src.attached_hat)
+		to_chat(user, span_notice("You removed [attached_hat.name] from helmet!"))
+		attached_hat = null
+		update_appearance()
+	return SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN
```

Comment:
Make 160 just ..() and then . = SECONDARY_blah

then you can inverse check on current 161, early return, and unindent 162 to 165

if there’s not a reason to specify src please remove those as well, might be missing something idk frick mobile.